dvc/repo/artifacts.py

```diff
@@ -50,18 +64,29 @@ def read(self) -> Dict[str, Dict[str, Artifact]]:
             dvcfile,
             dvcfile_artifacts,
         ) in self.repo.index._artifacts.items():  # pylint: disable=protected-access
-            # read the artifacts.yaml file if needed
-            if isinstance(dvcfile_artifacts, str):
-                dvcfile_artifacts = ArtifactsFile(
-                    self.repo,
-                    os.path.join(os.path.dirname(dvcfile), dvcfile_artifacts),
-                    verify=False,
-                ).load()
-            if not dvcfile_artifacts:
-                continue
             dvcyaml = relpath(dvcfile, self.repo.root_dir)
             artifacts[dvcyaml] = {}
             for name, value in dvcfile_artifacts.items():
                 check_name_format(name)
                 artifacts[dvcyaml][name] = Artifact(**value)
         return artifacts
+
+    def add(self, name: str, artifact: Artifact, dvcfile: Optional[str] = None):
+        # this doesn't update it "in place": self.read() won't return the updated value
+        with self.repo.scm_context(quiet=True):
+            if not name_is_compatible(name):
+                raise InvalidArgumentError(wrong_name_message % name)
```

Comment:
I'll suggest changing `check_name_format()` to raise an exception, and use it instead. 

You can catch that exception in `Artifacts.read()` and warn and suppress.

Reply:
Thanks @skshetry. Appreciate your review, made the changes.

Reply:
Did you forget to push?

Reply:
pushed it to another branch first 😅 pushed it here now!